src/astra_rl/training/trainer.py

```diff
@@ -0,0 +1,163 @@
+"""
+trainer.py
+This is a high-level trainer that imports the utilities in
+harness.py to create a pushbutton interface for training.
+
+Of course it will make a bunch of decisions for you. If
+you don't want that, the use the interface in harness.py.
+"""
+
+import torch
+from typing import Generic
+from pydantic import BaseModel
+from torch.optim import Optimizer
+
+from astra_rl.training.harness import Harness
+from astra_rl.core.environment import Environment
+from astra_rl.core.algorithm import Algorithm
+from astra_rl.core.common import ActionT, StateT, Batch, Step
+
+
+class TrainingConfiguration(BaseModel):
```

Comment:
nit: Add docstrings to document each parameter (https://google.github.io/styleguide/pyguide.html#384-classes)